httpcli/server.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+//const defaultPort = "8080"
+
+type jsBody struct {
+	Name string `json:"key" validate:"required"`
+	Age  int    `json:"age" validate:"required"`
+}
+
+type arrayFlags []string
+
+func (i *arrayFlags) String() string {
+	return "my string representation"
+}
+
+func (i *arrayFlags) Set(value string) error {
+	*i = append(*i, value)
+	return nil
+}
+
+func main() {
+	if len(os.Args) == 2 {
+		// helpCmd := flag.String("help", "", "provide http sub-commands")
+		helpPtr := flag.Bool("help", false, "help")
+		flag.Parse()
+
+		if *helpPtr {
+			fmt.Println("default command is http GET")
+			fmt.Println("you can specify GET PUSH PUT DELETE for other purposes")
+		} else {
+			resp, e := http.Get("https://" + os.Args[1])
+			if e != nil {
+				log.Fatal(e)
+			} else {
+				fmt.Println(resp)
+			}
+		}
+	} else {
+		getCmd := flag.NewFlagSet("get", flag.ExitOnError)
+		getHelp := getCmd.Bool("help", false, "help")
+		postCmd := flag.NewFlagSet("post", flag.ExitOnError)
+		postHelp := postCmd.Bool("help", false, "help")
+		postJSON := postCmd.Bool("json", false, "json")
+		deleteCmd := flag.NewFlagSet("delete", flag.ExitOnError)
+		deleteHelp := deleteCmd.Bool("help", false, "help")
+		putCmd := flag.NewFlagSet("put", flag.ExitOnError)
+		putHelp := putCmd.Bool("help", false, "help")
+
+		switch os.Args[1] {
+		case "get":
```

Comment:
we could get a cleaner main by calling a function in each switch case statement
the example would look like this

```go
case "get":
    util.Get(<necessary args>)
case "post":
    util.Post(<necessary args>)
...
```

Reply:
By doing this you will have to create a new directory and package `util` and imports the functions here